lib/types.ts

```diff
@@ -16,22 +16,26 @@ export type SegmentFilter = {
 export type AttributeFilter = {
   field: string;
   operator: FilterOperator;
-  value?: string;
+  value?: string | number | boolean;
 };
 
 export type NotFilter = {
   segment?: SegmentFilter;
   attribute?: AttributeFilter;
 };
 
-export type FilterObject = SegmentFilter | AttributeFilter | NotFilter;
+export type FilterObject = SegmentFilter | AttributeFilter;
+
+export type NotFilterObject = {
```

Comment:
This can replace `NotFilter` above since it's unused now. Otherwise this looks much better, thank you!

Reply:
I can't because it is still used in the docs, however It should probably be renamed "SegmentAndAttribute". What do you think ?

Reply:
Which docs? If it's in the README in this repo, it can be removed/fixed

Reply:
The official Customer.io docs. On [this page](https://www.customer.io/docs/api/#operation/getPeopleFilter), in the NodeJS example section, you'll see a use case that I have put in the tests. And the _'old'_ **NotFilter** can be used in every context (and, or, not)

Reply:
I actually think our documentation is wrong, and the structure if invalid. If you make a request with the example filters, the request will fail. I'm talking with the documentation team and will come back here once we figure it out. Thanks for your patience!

Reply:
Oh right ! Didn't tested the query, my bad ! Will update it, thank you :) !